test/functional/api/v2/hostgroups_controller_test.rb

```diff
@@ -63,6 +63,32 @@ class HostgroupsControllerTest < ActionController::TestCase
         assert_response :success
         assert_equal assigns('hostgroup').ansible_roles, [@ansible_role2, @ansible_role1]
       end
+
+      test 'should append a role to a hostgroup with a correct ordering' do
+        hostgroup = FactoryBot.create(:hostgroup,
+                                      :ansible_role_ids => [@ansible_role1.id])
+        post :ansible_roles,
+             :params => {
+               :id => hostgroup.id,
+               :ansible_role_id => [@ansible_role2.id]
+             },
+             :session => set_session_user
+        assert_response :success
```

Comment:
I'd expect HTTP 201 Created if it was added, not HTTP 200 Success

Reply:
The server responds with correct `201`.

P.S. Apparently `post` and others are expecting `:action` instead of route from `routes` definition. `ofedoren.experience += 1`. Also, `assert_response` here considers both `200` and `201` as successful via mapping, that's why the test doesn't fail, but I've changed to use the specific number instead.
P.P.S. I hate test frameworks :/